src/device/cuda/wmma.jl

```diff
@@ -103,7 +126,11 @@ for mat in ["a", "b", "c"],
 
     ccall_name = "extern $llvm_intr"
 
+    if addrspaceptr_available
+    @eval $func_name(src_addr, stride) = ccall($ccall_name, llvmcall, NTuple{$sz, $frag_ty}, (Core.AddrSpacePtr{$arr_ty, $addr_space_int}, Int32), convert(Core.AddrSpacePtr{$arr_ty, $addr_space_int}, src_addr), stride)
```

Comment:
You can avoid these multiple definitions by implementing `cconvert` instead; that method is called already. Furthermore, using a variable for the destination type(Ref or AddrSpacePtr) you can get rid of the `if` too.